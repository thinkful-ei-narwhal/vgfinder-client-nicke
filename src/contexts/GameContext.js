import React, { Component } from "react";

export const nullGame = {
  id: "",
  title: "",
  description: "",
  genre: "",
  rating: "",
  release_date: "",
  developer: "",
  trailer_url: "",
  image_url_box_art: "",
  image_url_two: "",
  image_url_three: "",
  image_url_four: "",
  image_url_five: "",
};

const GameContext = React.createContext({
  game: nullGame,
  setGame: () => {},
  clearGame: () => {},
});

export default GameContext;

export class GameProvider extends Component {
  state = {
    game: nullGame,
  };

  setGame = (game) => {
    this.setState({ game });
  };

  clearGame = () => {
    this.setGame(nullGame);
  };

  render() {
    const value = {
      game: this.state.game,
      setGame: this.setGame,
      clearGame: this.clearGame,
    };
    return (
      <GameContext.Provider value={value}>
        {this.props.children}
      </GameContext.Provider>
    );
  }
}
