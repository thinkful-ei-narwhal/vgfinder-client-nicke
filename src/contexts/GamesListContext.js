import React, { Component } from 'react';

const GamesListContext = React.createContext({
  gamesList: [],
  error: null,
  setError: () => { },
  clearError: () => { },
  setGamesList: () => { },
})
export default GamesListContext

export class GamesListProvider extends Component {
  state = {
    gamesList: [],
    error: null,
  };

  setGamesList = gamesList => {
    this.setState({ gamesList })
  }

  setError = error => {
    console.error(error)
    this.setState({ error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  render() {
    const value = {
      gamesList: this.state.gamesList,
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setGamesList: this.setGamesList,
    }
    return (
      <GamesListContext.Provider value={value}>
        {this.props.children}
      </GamesListContext.Provider>
    )
  }
}
