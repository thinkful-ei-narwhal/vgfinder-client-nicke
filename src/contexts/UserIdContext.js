import React, { Component } from 'react'

const UserIdContext = React.createContext({
  userId: undefined,
  wishlist: [],
  setUserId: () => { },
  setUserWishlist: () => { },
})

export default UserIdContext

export class UserIdContextProvider extends Component {
  state = {
    userId: undefined,
    wishlist: []
  };

  setUserId = (userId) => {
    this.setState({ userId })
  }

  setUserWishlist = (wishlist) => {
    this.setState({ wishlist })
  }


  render() {
    const value = {
      userId: this.state.userId,
      wishlist: this.state.wishlist,
      setUserId: this.setUserId,
      setUserWishlist: this.setUserWishlist
    }

    return (
      <UserIdContext.Provider value={value}>
        {this.props.children}
      </UserIdContext.Provider>
    )
  }
}
