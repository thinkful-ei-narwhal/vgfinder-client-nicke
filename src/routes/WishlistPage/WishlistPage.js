import './WishlistPage.css'
import React, { Component } from 'react';
import Header from '../../components/Header/Header';
import { Section } from '../../components/Utils/Utils'
import UserIdContext from '../../contexts/UserIdContext';
import Search from '../../components/Search/Search'
import WishlistApiService from '../../services/wishlist-api-service';

export default class WishlistPage extends Component {
  static contextType = UserIdContext;

  constructor(props) {
    super(props)
    this.state = {
      loaded: false
    }
  }

  componentDidMount() {
    const userId = this.context.userId
    let userWishlist = [];
    let userGamesList = [];

    //get all wishlists for user
    const wishlistPromise1 = WishlistApiService.getAllWishlists()
      .then(res => {
        userWishlist = res.filter(wishlist => wishlist.user_id === userId);
      })
      .catch(this.context.setError)

    //get all wishlisted game data
    const wishlistPromise2 = WishlistApiService.getwishlistedGames(userId)
      .then(res => {
        userGamesList = res;
      })
      .catch(this.context.setError)

    Promise.all([wishlistPromise1, wishlistPromise2]).then(() => {
      this.context.setUserIdWishlistAndGames(userId, userWishlist, userGamesList);
      this.setState({ loaded: true })
    });
  }

  renderWishlist() {
    return <>
      <Section className='GamePage'>
        <Search gamesList={this.context.userGames} />
      </Section>
    </>
  }

  render() {
    return (
      <>
        <Header pathName={this.props.location.pathname} />
        {this.state.loaded ? this.renderWishlist() : null}
      </>
    )
  }
}