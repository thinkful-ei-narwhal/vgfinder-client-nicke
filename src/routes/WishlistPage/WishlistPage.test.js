import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import WishlistPage from "./WishlistPage";
import { UserIdContextProvider } from "./../../contexts/UserIdContext";

describe(`WishlistPage component`, () => {
  it("renders a WishlistPage component", () => {
    const wrapper = shallow(
      <UserIdContextProvider>
        <WishlistPage />
      </UserIdContextProvider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
