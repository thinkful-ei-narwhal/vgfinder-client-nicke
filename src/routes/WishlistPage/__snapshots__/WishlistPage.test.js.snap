// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WishlistPage component renders a WishlistPage component 1`] = `
<ContextProvider
  value={
    Object {
      "setUserId": [Function],
      "setUserIdWishlistAndGames": [Function],
      "userGames": Array [],
      "userId": undefined,
      "wishlist": Array [],
    }
  }
>
  <WishlistPage />
</ContextProvider>
`;
